src/Features/Core/Portable/Snippets/SnippetProviders/AbstractSnippetProvider.cs

```diff
@@ -124,5 +202,10 @@ private async Task<Document> AddFormatAnnotationAsync(Document document, int pos
             document = document.WithSyntaxRoot(annotatedSnippetRoot);
             return document;
         }
+
+        protected virtual ImmutableArray<RoslynLSPSnippetItem> GetPlaceHolderLocationsList(SyntaxNode node, ISyntaxFacts syntaxFacts, CancellationToken cancellationToken)
+        {
+            return ImmutableArray<RoslynLSPSnippetItem>.Empty;
+        }
```

Comment:
fwiw, my preference is that thsi method is abstract, and all snippets need to think about thsi (even if they return empty).